source/common/network/io_socket_handle_impl.h

```diff
@@ -0,0 +1,42 @@
+#pragma once
+
+#include "envoy/network/io_handle.h"
+
+namespace Envoy {
+namespace Network {
+
+/**
+ * IoHandle derivative for sockets
+ */
+class IoSocketHandle : public IoHandle {
+public:
+  IoSocketHandle(int fd = -1) : fd_(fd) {}
+
+  ~IoSocketHandle() {}
```

Comment:
Mind adding a TODO to close() on destruction?
Now that we have a wrapper class it'd be good to never leak fds, but I'm prefer this be as pure-refactor as possible and I could imagine that causing functional change if we're not careful.

Reply:
Done.

Reply:
For now, let's at least `ASSERT(fd_ == -1);` in the destructor.

Reply:
Good point.